### PR TITLE
[DNM] add epoch to command request

### PR DESCRIPTION
**Pull request by BusyJay**
@ngaut @siddontang @morefreeze

### CODE REVIEW

File: proto/errorpb.proto [hunk 1]
```diff
@@ -19,9 +19,13 @@ message KeyNotInRegion {
...
 
+message EpochOutdated {
+}
+
 message Error {
     optional string message                         = 1;
     optional NotLeaderError not_leader              = 2;
     optional RegionNotFoundError region_not_found   = 3;
     optional KeyNotInRegion key_not_in_region       = 4;
+    optional EpochOutdated epoch_outdated           = 5;
```
Comment (siddontang) [inline proto/errorpb.proto]: I prefer stale_epoch

@ngaut

Comment (BusyJay) [inline proto/errorpb.proto]: When user tell if an error is `EpochOutdated`, he will write `has_epoch_outdated`, which just sound like a normal question: "Has epoch (be) outdated?"

Comment (siddontang) [inline proto/errorpb.proto]: got it

Comment (siddontang) [inline proto/errorpb.proto]: Then I prefer epoch_stale, we use stale commonly.  :smile:

**Comment by BusyJay** [comment]
@ngaut @siddontang PTAL

File: proto/raft_cmdpb.proto [hunk 2]
```diff
@@ -169,12 +169,14 @@ message StatusResponse {
...
+    optional uint64 region_id                   = 1;
+    optional metapb.Peer peer                   = 2;
     // true for read linearization
-    optional bool read_quorum          = 3;
+    optional bool read_quorum                   = 3;
     // 16 bytes, to distinguish request.  
-    optional bytes uuid                = 4;
+    optional bytes uuid                         = 4;
+    
+    optional metapb.RegionEpoch region_epoch    = 5;
```
Comment (ngaut) [inline proto/raft_cmdpb.proto]: Do we really need this field, Most of RaftCmdRequest already have the field?

**Comment by siddontang** [comment]
LGTM

**Comment by morefreeze** [comment]
I don't know why add this. Can you talk detail or @ngaut take a look.

**Comment by siddontang** [comment]
You must send this epoch in command, and we will use this to check whether your client has a stale region or not.

**Comment by siddontang** [comment]
Maybe we can return epoch in every response too. @ngaut @BusyJay

**Comment by BusyJay** [comment]
For what?

**Comment by siddontang** [comment]
oh, no need, if the epoch is stale, an error returns.

**Comment by ngaut** [comment]
LGTM

**Pull request merged**